### PR TITLE
Skip signing tests for insufficient credentials

### PR DESCRIPTION
**Pull request by crwilcox**

### CODE REVIEW

**Comment by crwilcox** [comment]
Fix for internal bug: 113623918

File: storage/tests/system.py [hunk 1]
```diff
@@ -699,6 +700,10 @@ def test_third_level(self):
         self.assertEqual(iterator.prefixes, set())
 
 
+@unittest.skipUnless(
+    type(storage.Client()._credentials) is google.oauth2.service_account.credentials,
+    "Can only test signing with credentials capable of signing",
+)
```
Comment (tseaver) [inline storage/tests/system.py]: We can't instantiate a client at import time here, because there might not be *any* credentials.  What is the usecase?

Comment (crwilcox) [reply storage/tests/system.py]: If a user creates their credentials through gCloud they will have credentials of the type google.oauth2.client. These credentials are not capable of singing and these tests will fail for no reason other than that

Comment (tseaver) [reply storage/tests/system.py]: We need to raise the skip from inside the testcase, then.

File: storage/tests/system.py [hunk 2]
```diff
@@ -703,6 +704,12 @@ class TestStorageSignURLs(TestStorageFiles):
     def setUp(self):
         super(TestStorageSignURLs, self).setUp()
 
+        if (
+            type(storage.Client()._credentials)
```
Comment (tseaver) [inline storage/tests/system.py]: Inside here, you should be able to use `Config.CLIENT._credentials`.

**Pull request merged**